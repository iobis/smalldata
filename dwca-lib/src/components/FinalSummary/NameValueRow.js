import classNames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'

export default function NameValueRow({ className, name, value }) {
  return (
    <tr className={classNames('name-value-row fieldrow', className)}>
      <td className="name">{name}</td>
      <td className="value">{!value ? 'â€”' : value}</td>
    </tr>
  )
}

NameValueRow.propTypes = {
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  name:      PropTypes.string.isRequired,
  value:     PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}
