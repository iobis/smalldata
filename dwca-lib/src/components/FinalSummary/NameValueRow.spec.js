import NameValueRow from './NameValueRow'
import React from 'react'
import { mount } from 'enzyme'

describe('NameValueRow', () => {
  it('renders correctly for string value', () => {
    expect(mount(
      <table>
        <tbody>
          <NameValueRow name="name" value="value string"/>
        </tbody>
      </table>
    )).toMatchSnapshot()
  })

  it('renders correctly for numeric value', () => {
    expect(mount(
      <table>
        <tbody>
          <NameValueRow name="name" value={12345}/>
        </tbody>
      </table>
    )).toMatchSnapshot()
  })

  it('renders correctly if className is provided', () => {
    expect(mount(
      <table>
        <tbody>
          <NameValueRow className="custom-class-name" name="name" value={12345}/>
        </tbody>
      </table>
    )).toMatchSnapshot()
  })
})
