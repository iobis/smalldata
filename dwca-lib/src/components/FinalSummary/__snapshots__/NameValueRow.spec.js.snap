// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameValueRow renders correctly for numeric value 1`] = `
<table>
  <tbody>
    <NameValueRow
      name="name"
      value={12345}
    >
      <tr
        className="name-value-row fieldrow"
      >
        <td
          className="name"
        >
          name
        </td>
        <td
          className="value"
        >
          12345
        </td>
      </tr>
    </NameValueRow>
  </tbody>
</table>
`;

exports[`NameValueRow renders correctly for string value 1`] = `
<table>
  <tbody>
    <NameValueRow
      name="name"
      value="value string"
    >
      <tr
        className="name-value-row fieldrow"
      >
        <td
          className="name"
        >
          name
        </td>
        <td
          className="value"
        >
          value string
        </td>
      </tr>
    </NameValueRow>
  </tbody>
</table>
`;

exports[`NameValueRow renders correctly if className is provided 1`] = `
<table>
  <tbody>
    <NameValueRow
      className="custom-class-name"
      name="name"
      value={12345}
    >
      <tr
        className="name-value-row fieldrow custom-class-name"
      >
        <td
          className="name"
        >
          name
        </td>
        <td
          className="value"
        >
          12345
        </td>
      </tr>
    </NameValueRow>
  </tbody>
</table>
`;
